### DISCUSSION TITLE
RNS 0.3.14 Released - Update All Applications

### DISCUSSION THREAD

**Discussion by markqvist** [markqvist/Reticulum#112] [Announcements]
This maintenance release brings a few improvements, including optimised announce packet structure and updated documentation.

**Please note!** While this is a small maintenance release, it includes changes to packe structure that breaks backwards compatibility with all previous RNS versions.

**Changes**
- Optimised announce packet structure
- Reject mismatching public keys on hash collision.
- Minor updates to documentation

**Release Hashes**
```
SHA256 b761efc24d20c5719817bfefbbe8ce69f7c91d65bb8273cb02578f77d6f88bc5 rns-0.3.14-py3-none-any.whl
SHA256 cc24a1f010431c8f193ec0ffc6dccade614a5be40c47ac12e3e9ae60b52f046e rnspure-0.3.14-py3-none-any.whl
```

**Comment by markqvist** [comment]
[LXMF](https://github.com/markqvist/LXMF/releases/tag/0.2.0), [Nomad Network](https://github.com/markqvist/NomadNet/releases/tag/0.2.4) and [Sideband](https://github.com/markqvist/Sideband/releases/tag/0.2.1) have also been updated to use the latest RNS release.

This release, *again*, intentionally breaks backwards compatibility with all previous version. I know it has been probably been a little annoying with three updates like this in a very short amount of time, but in the end it was really necessary to do, and I figured it would be better to just get it over with. At this point I am quite confident we won't have any more of these for a long time, and that future versions should maintain good compatibility across versions. These sort of "breaks" only happen when the wire-format is changed, which I had to do this time around.

Due to the changes implemented in the announce packet structure and new announce checks, *all adresses will change*. You can keep your existing identities, but LXMF and node address hashes will change in 0.3.14.

I highly recommend updating your systems to this release as soon as possible. The testnet has already been updated.

**Comment by moonlit-tune** [comment]
@markqvist Will there be a source distribution for nomadnet 0.2.4 published to PyPI? I only see a wheel as of now. Arch's convention for Python packages is to use source distributions where available; just checking so I know whether to switch to using the source from Github. Thanks!

**Reply by markqvist** [reply]
Eer, if that is the case, it is a mistake, something funky must have happened while uploading to pypi. I get it fixed in a snap.

**Reply by markqvist** [reply]
Ok my fault, the `nomadnet-0.2.4-1.tar.gz` source distribution should be there now.

**Reply by moonlit-tune** [reply]
Got it, thanks!
